common/src/main/java/com/netflix/conductor/common/metadata/workflow/SubWorkflowParams.java

```diff
@@ -100,31 +96,28 @@ public Object getWorkflowDefinition() {
         return workflowDefinition;
     }
 
-    /**
-     * @return the workflowDefinition as a WorkflowDef
-     */
-    @JsonGetter("workflowDefinition")
-    public WorkflowDef getWorkflowDef() {
-        return (WorkflowDef) workflowDefinition;
-    }
-
     /**
      * @param workflowDef the workflowDefinition to set
      */
     public void setWorkflowDefinition(Object workflowDef) {
-        if (!(workflowDef == null || workflowDef instanceof WorkflowDef)) {
+        if (workflowDef == null) {
+            this.workflowDefinition = workflowDef;
+        } else if (workflowDef instanceof WorkflowDef) {
+            this.workflowDefinition = workflowDef;
+        } else if (workflowDef instanceof String) {
+            if (!(((String) workflowDef).startsWith("${"))
```

Comment:
Thought to make it a regex but after thinking more about it - this is fine, there is some room in what is allowed as a valid DSL string for parameter utils